binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,53 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        def go(value, curr):
+            if value > curr.value:
+                if curr.right is None:
+                    curr.right = BinarySearchTree(value)
+                    return
+                else:
+                    go(value, curr.right)
+            elif value < curr.value:
+                if curr.left is None:
+                    curr.left = BinarySearchTree(value)
+                    return
+                else:
+                    go(value, curr.left)
+        go(value, self)
 
```

Comment:
very interesting recursive implementation; think it could have been done without the nested method. I like the clear code overall.

Reply:
you have an elif but no else statement; elif could either be changed to be else since you are checking if value > curr.value, the else would then check for the other case.

Reply:
should not need to change what you already have but would add an else to catch any errors that might occur